packages/headless/src/features/search/search-actions-loader.ts

```diff
@@ -55,6 +57,20 @@ export interface SearchActionCreators {
       ClientThunkExtraArguments<SearchAPIClient>
     >
   >;
+
+  /**
+   * Prepares the search state for a search with a query by setting the query string and resetting facet and pager states.
+   *
+   * @param q (string) The new basic query expression (e.g., `acme tornado seeds`).
```

Comment:
I think you will see the failed build in Jenkins due to a mismatch between the JSDOC and the actual signature of the function (with the `payload` parameter).

You can run `npm run doc:generate` locally at the top of `packages/headless` to see if it pass or fails.